src/store.js

```diff
@@ -9,19 +9,69 @@ class Store {
 
   static addList(list) {
     const lists = Store.getList();
+    if (lists.length === 0) {
+      list.id = 1;
+    } else {
+      const lastList = lists[lists.length - 1];
+      list.id = lastList.id + 1;
+    }
     lists.push(list);
     localStorage.setItem('list', JSON.stringify(lists));
   }
 
+  static updateList(list) {
+    const lists = Store.getList();
+    const index = lists.findIndex((item) => item.id === list.id);
+    if (index === -1) {
+      if (lists.length === 0) {
+        list.id = 1;
+      } else {
+        const lastList = lists[lists.length - 1];
+        list.id = lastList.id + 1;
+      }
+      lists.push(list);
+    } else {
+      lists[index] = list;
+    }
+    localStorage.setItem('list', JSON.stringify(lists));
+  }
+
+  static updateCompleted(list) {
+    const lists = Store.getList();
+    const index = lists.findIndex((item) => item.id === list.id);
+
+    if (index !== -1) {
+      lists[index].completed = !lists[index].completed;
+      localStorage.setItem('list', JSON.stringify(lists));
+    }
+  }
+
+  static uncheckCompleted(list) {
+    const lists = Store.getList();
+    const index = lists.findIndex((item) => item.id === list.id);
+
+    if (index !== -1) {
+      lists[index].completed = false;
+      localStorage.setItem('list', JSON.stringify(lists));
+    }
+  }
+
   static removeList(id) {
     const lists = Store.getList();
     const index = lists.findIndex((list) => list.id === id);
-
     if (index !== -1) {
       lists.splice(index, 1);
       localStorage.setItem('list', JSON.stringify(lists));
     }
   }
```

Comment:
- [ ] When I tried to remove a task the whole application freezes and doesn't respond anymore so please make sure to fix this issue to complete the requirements 
<img width="1440" alt="Screenshot 2023-05-12 at 18 48 35" src="https://github.com/prg-04/To-Do_List/assets/61790587/98de3a9c-e2d0-449d-88df-77d67f770f27">